clients/include/rocblas.hpp

```diff
@@ -28,23 +28,27 @@
  */
 
 // scal
-template <typename T>
+template <typename T, typename U = T>
 rocblas_status (*rocblas_scal)(
-    rocblas_handle handle, rocblas_int n, const T* alpha, T* x, rocblas_int incx);
+    rocblas_handle handle, rocblas_int n, const U* alpha, T* x, rocblas_int incx);
 
 template <>
 static constexpr auto rocblas_scal<float> = rocblas_sscal;
 
 template <>
 static constexpr auto rocblas_scal<double> = rocblas_dscal;
 
-/* not implemented
 template <>
 static constexpr auto rocblas_scal<rocblas_float_complex> = rocblas_cscal;
 
 template <>
 static constexpr auto rocblas_scal<rocblas_double_complex> = rocblas_zscal;
-*/
+
+template <>
+static constexpr auto rocblas_scal<rocblas_float_complex, float> = rocblas_csscal;
```

Comment:
The types should be switched here:
`static constexpr auto rocblas_scal<float, rocblas_float_complex> = rocblas_csscal;`
As well as for double types.